src/anyio/_backends/_asyncio.py

```diff
@@ -314,20 +321,23 @@ def shield(self) -> bool:
         return self._shield
 
 
-async def checkpoint():
+def _cancel_called(task):
     try:
-        cancel_scope = _task_states[current_task()].cancel_scope
+        cancel_scope = _task_states[task].cancel_scope
     except KeyError:
         cancel_scope = None
 
     while cancel_scope:
         if cancel_scope.cancel_called:
-            raise CancelledError
-        elif cancel_scope.shield:
-            break
-        else:
-            cancel_scope = cancel_scope._parent_scope
+            return True
+        if cancel_scope.shield:
+            return False
+        cancel_scope = cancel_scope._parent_scope
```

Comment:
Empty lines after control blocks.
```suggestion

        if cancel_scope.shield:
            return False

        cancel_scope = cancel_scope._parent_scope
```